Mathlib/Data/Sum/Basic.lean

```diff
@@ -47,29 +44,29 @@ theorem inr_injective : Function.Injective (inr : β → α ⊕ β) := fun _ _ 
 
 theorem sum_rec_congr (P : α ⊕ β → Sort*) (f : ∀ i, P (inl i)) (g : ∀ i, P (inr i))
     {x y : α ⊕ β} (h : x = y) :
-    @Sum.rec _ _ _ f g x = cast (congr_arg P h.symm) (@Sum.rec _ _ _ f g y) := by cases h; rfl
+    @Sum.rec _ _ _ f g x = cast (congr_arg P h.symm) (@Sum.rec _ _ _ f g y) := by grind
```

Comment:
I'd argue the original proof was better, as it uses simpler tactics and is already very short as is.